javatests/arcs/core/allocator/AllocatorTest.kt

```diff
@@ -1,134 +1,119 @@
-class AllocatorTest {
+open class AllocatorTest {
     /**
-     * A test recipe, two particles
-     * [WritePerson] writes a Person to a handle
-     * [ReadPerson] reads a Person from a handle
-     * [WritePerson] runs in [WritingHost]
-     * [ReadPerson] runs in [ReadingHost]
-     *
-     * Hand translated 'compilation' roughly equivalent to:
-     *
-     * schema Person { name: Text }
-     * particle ReadPerson in 'arcs.core.host.AllocatorTest.ReadPerson'
-     *   person: reads Person
-     *
-     * particle WritePerson in 'arcs.core.host.AllocatorTest.WritePerson'
-     *   person: writes Person
-     *
-     * recipe WriteAndReadPerson
-     *   recipePerson: create
-     *   WritePerson
-     *     person: writes recipePerson
-     *   ReadPerson
-     *     person: reads recipePerson
+     * Recipe hand translated from 'person.arcs'
      */
+    private lateinit var recipePersonStorageKey: StorageKey
+    private lateinit var allocator: Allocator
+    private lateinit var hostRegistry: HostRegistry
+    private lateinit var readPersonHandleConnection: Plan.HandleConnection
+    private lateinit var writePersonHandleConnection: Plan.HandleConnection
+    private lateinit var writePersonParticle: Plan.Particle
+    private lateinit var readPersonParticle: Plan.Particle
+    private lateinit var writeAndReadPersonPlan: Plan
     private val personSchema = Schema(
         listOf(SchemaName("Person")),
         SchemaFields(mapOf("name" to FieldType.Text), emptyMap()),
-        "hash"
+        "42"
     )
+    private var personEntityType: Type = SingletonType(EntityType(personSchema))
 
-    private val personEntityType = EntityType(personSchema)
-
-
-    class WritePerson : Particle
-    class ReadPerson : Particle
-
-    open class TestingHost(vararg particles: KClass<out Particle>) :
-        AbstractArcHost(particles.toIdentifierList()) {
-
-        var started = mutableListOf<Plan.Partition>()
-
-        override suspend fun startArc(partition: Plan.Partition) {
-            super.startArc(partition)
-            started.add(partition)
-        }
-
-        fun setup() {
-            started.clear()
-        }
-    }
+    private lateinit var readingExternalHost: TestingHost
+    private lateinit var writingExternalHost: TestingHost
 
     class WritingHost : TestingHost(WritePerson::class)
     class ReadingHost : TestingHost(ReadPerson::class)
 
-    private lateinit var readPersonHandleConnection: Plan.HandleConnection
-    private lateinit var writePersonHandleConnection: Plan.HandleConnection
-    private lateinit var writePersonParticle: Plan.Particle
-    private lateinit var readPersonParticle: Plan.Particle
-    private lateinit var writeAndReadPersonPlan: Plan
-    private lateinit var hostRegistry: ExplicitHostRegistry
+    open fun readingHost(): TestingHost = ReadingHost()
+    open fun writingHost(): TestingHost = WritingHost()
+    open fun storageCapability() = Capabilities.TiedToRuntime
```

Comment:
Would be good to add KDoc for these, also - can they be `val`s?

Reply:
Added KDoc. I think they need to be fun because they can be late initialized.

Reply:
if they are `open val`, the implementer could always use the `lazy` delegate.

Reply:
What I mean is, there's a new copy of these made everytime setUp() is called